Add content that explains how to override Dapr's default retries (#2590)
dapr/docs: daprdocs/content/en/operations/resiliency/policies.md

| Actors                 | DaprBuiltInActorRetries          | Per call retries are performed with a backoff interval of 1 second, up to a threshold of 3 times.                                                                                     | Sidecar-to-sidecar requests (an actor method call) that fail and result in a gRPC code `Unavailable` or `Unauthenticated`              |
| Actor Reminders        | DaprBuiltInActorReminderRetries  | Per call retries are performed with an exponential backoff with an initial interval of 500ms, up to a maximum of 60s for a duration of 15mins                                 | Requests that fail to persist an actor reminder to a state store                                                                      | 
| Initialization Retries | DaprBuiltInInitializationRetries | Per call retries are performed 3 times with an exponential backoff, an initial interval of 500ms and for a duration of 10s                                                       | Failures when making a request to an application to retrieve a given spec. For example, failure to retrieve a subscription, component or resiliency specification       | 

> Example of overriding default retries for Service Invocation:

Author: ```suggestion
Below is an example of a Resiliency spec that includes overriding default retries for all service invocation requests by using the reserved, named keyword 'DaprBuiltInServiceRetries'. Also defined is a custom retry policy called 'retryForever' that is only applied to the appB target. 

appB will use the 'retryForever' retry policy, while all other service invocations failures will default to using the overridden 'DaprBuiltInServiceRetries' default policy.
```

Author: @msfussell, does this help clarify what the resiliency spec example demonstrates?

Reviewer: I suggest this

```suggestion
The resiliency spec example below shows overriding the default retries for _all_ service invocation requests by using the reserved, named keyword 'DaprBuiltInServiceRetries'. 

Also defined is a retry policy called 'retryForever' that is only applied to the appB target.  appB uses the 'retryForever' retry policy, while all other application service invocation retry failures, use the overridden 'DaprBuiltInServiceRetries' default policy.
```

```yaml
spec:
  policies:
    retries:
      DaprBuiltInServiceRetries: # Overrides default retry behaivor for service-to-service calls
        policy: constant
        duration: 5s
        maxRetries: 10

>       retryForever: # A custom retry policy will replace default retries and the target will rely soley on the named policy. 

Author: ```suggestion
      retryForever: # A user defined retry policy replaces default retries. Targets rely solely on the applied policy. 
```

Reviewer: Can you remove the "will"s in this and use the present tense?

        timeout: 45s
        trip: consecutiveFailures > 8
```

##### Override Default Retries

> Dapr provides default retries for certain request failures and transient errors.  Within a resiliency spec, you have the option to override Dapr's default retry logic by defining policies with reserved, named keywords (for example, defining a policy with the name `DaprBuiltInServiceRetries`, will override default retries for failures between sidecars via service-to-service requests). Policy overrides are not applied to specific targets. 

Reviewer: ```suggestion
Dapr provides default retries for certain request failures and transient errors.  Within a resiliency spec, you have the option to override Dapr's default retry logic by defining policies with reserved, named keywords. For example, defining a policy with the name `DaprBuiltInServiceRetries`, overrides the default retries for failures between sidecars via service-to-service requests. Policy overrides are not applied to specific targets. 
```

Reviewer: Re "Policy overrides are not applied to specific targets.". Is this saying "The default policy overrides are not applied to specific targets, since they have a target already specified."?

Author: @msfussell no, this implies that when you override a default policy (using a reserved, name keyword) Dapr knows where to apply that policy. You do not need to add that policy to any targets in the target section below

> Note: Although you can override default values with more robust retries, you cannot override with lesser values than the provided default value, or completely remove default retries. This prevents unexpected downtime. 

Below is a table that describes Dapr's default retries and the policy keywords to override them: 

| Capability             | Override Keyword                 | Default Retry Behavior                                                                                                                                                               | Description                                                                                                                           |
| ------------------     | -------------------------        | ------------------------------                                                                                                                                                       | -----------------------------------------------------------------------------------------------------------                           |

> | Service Invocation     | DaprBuiltInServiceRetries        | Per call retries are performed with a backoff interval of 1 second up to a threshold of 3 times.                                                                                     | Sidecar-to-sidecar requets (via a service invocation call) that fails and results in a gRPC code `Unavailable` or `Unauthenticated`   | 

Reviewer: ```suggestion
| Service Invocation     | DaprBuiltInServiceRetries        | Per call retries are performed with a backoff interval of 1 second, up to a threshold of 3 times.                                                                                     | Sidecar-to-sidecar requests (a service invocation method call) that fail and result in a gRPC code `Unavailable` or `Unauthenticated`   | 
```

> | Actors                 | DaprBuiltInActorRetries          | Per call retries are performed with a backoff interval of 1 second up to a threshold of 3 times.                                                                                     | Sidecar-to-sidecar requests (to a remote actor) that fails and results in a gRPC code `Unavailable` or `Unauthenticated`              |

Reviewer: ```suggestion
| Actors                 | DaprBuiltInActorRetries          | Per call retries are performed with a backoff interval of 1 second, up to a threshold of 3 times.                                                                                     | Sidecar-to-sidecar requests (an actor method call) that fail and result in a gRPC code `Unavailable` or `Unauthenticated`              |
```

> | Actor Reminders        | DaprBuiltInActorReminderRetries  | Per call retries are performed with an exponential backoff wtih an initial interval of 500ms and up to a maximum of 60s for a duration of 15m                                        | Requests that fail to persist an actor reminder to a state store                                                                      | 

Reviewer: ```suggestion
| Actor Reminders        | DaprBuiltInActorReminderRetries  | Per call retries are performed with an exponential backoff with an initial interval of 500ms, up to a maximum of 60s for a duration of 15mins                                 | Requests that fail to persist an actor reminder to a state store                                                                      | 
```

> | Initialization Retries | DaprBuiltInInitializationRetries | Per call retries are performed 3 times wtih an exponential backoff with an initial interval of 500ms and for a duration of 10s                                                       | Failures when making a request to an application to retrieve a given spec. For example, failure to retrieve a Subscription spec       | 

Reviewer: ```suggestion
| Initialization Retries | DaprBuiltInInitializationRetries | Per call retries are performed 3 times with an exponential backoff, an initial interval of 500ms and for a duration of 10s                                                       | Failures when making a request to an application to retrieve a given spec. For example, failure to retrieve a subscription, component or resiliency specification       | 
```

> Example of overriding default retries for Service Invocation:

Reviewer: I would describe what this example is supposed to show, so then the reader has an expectation.

```yaml
spec:
  policies:

>     # Retries are named templates for retry configurations and are instantiated for life of the operation.

Reviewer: Why do you say templates here?  And why say "for the life of the operation"

Author: We can actually remove this comment

>     # Retries are named templates for retry configurations and are instantiated for life of the operation.

Author: ```suggestion
```

    retries:
      DaprBuiltInServiceRetries: # Overrides default retry behaivor for service-to-service calls
        policy: constant
        duration: 5s
        maxRetries: 10

>       retryForever: # A custom retry policy will replace default retries and the target will rely soley on the named policy. 

Reviewer: Is this just showing another retry policy? I find the comment on customer retry policy confusing here. What is custom about this and this has nothing to do with the default policies. Also default and BuiltIn are two words for the same thing and we should just pick one.

Author: When you define a retry policy and assign it to a target service. The target will only use that policy you've defined instead of any dapr default retries. 

Another way to say this is -  it's impossible to have default retries and custom retries stacked. You can only have one or the other at any time

        policy: exponential
        maxInterval: 15s

>         maxRetries: -1 # Retry indefinitely

Reviewer: DO we document -1 here? This is always strange IMO unless we are really clear on when to use indefinitely.

Author: @msfussell - Yes, it is defined in the retry section above:

| `maxRetries` | The maximum number of retries to attempt. `-1` denotes an indefinite number of retries. Defaults to `-1`. |
